src/content/posts/2024/08/26/swift-macros.mdx

```diff
@@ -0,0 +1,94 @@
+
+## What Tuist is doing
+
+Tuist is uniquely positioned to solve this problem thanks to our ability to optimize a source's dependency graph with binaries generated from previous builds. As soon as Swift Macros came out and we started seeing the build time issues, we extended [caching](https://docs.tuist.io/guides/develop/build/cache) to support Swift Macros too. Adopting this is very straightforward if you are using [Tuist Projects](https://docs.tuist.io/guides/develop/projects). All you need to do is run the following command to fingerprint and store your Swift Macros, frameworks, and bundles:
+
+```
+tuist cache
+```
+
+And the next time you or anyone on your team generates an Xcode project, they'll be using a previously-generated binary. It feels truly magical to see how fast the build times are after adopting this feature.
+
+If you want to see them in action, you can play with one of the Tuist project's fixtures:
+
+1. Clone the repository: `git clone https://github.com/tuist/tuist`.
+2. Install the repository dependencies: `mise install`.
+3. Choose the fixture directory: `cd fixtures/framework_with_native_swift_macro`
+4. Install the project dependencies: `tuist install`
+5. Cache the dependencies: `tuist cache`
+6. Generate the Xcode project: `tuist generate`
+
+The following image shows the generated Xcode project where all the dependencies, including the Swift Macros, have been cached:
+
+<Image src={cachedSwiftMacro} alt="An screenshot that shows an Xcode project where the file navigator includes the one source file contained in the framework, and all the external dependencies cached as XCFrameworks" style="max-width: 40vw" />
+
+## Closing
+
+There are three possible solutions we might see Apple adopting:
+
+1. Making SwiftSyntax ABI and API stable, and Swift ABI stable on non-Darwin platforms.
+2. Using WebAssembly to run Swift Macros.
+3. Supporting fingerprint-based binary caching in Xcode and Swift Package Manager.
+
+1 would be highly beneficial for the community, but it will test Apple's willingness to invest in other platforms or leverage the community to make that happen. 2 would show Apple's willingness to embrace web technologies, avoid reinventing the wheel, and help advance the web ecosystem, but Apple hasn't had a good record of doing that in the past. 3 would be the most pragmatic solution, but it would require Apple to make a significant investment in the build system, which has been historically slow to evolve, and there are many projects out there that accidentally build through implicit configuration that might break if such a feature is introduced.
```

Comment:
```suggestion
1) would be highly beneficial for the community, but it will test Apple's willingness to invest in other platforms or leverage the community to make that happen. 2) would show Apple's willingness to embrace web technologies, avoid reinventing the wheel, and help advance the web ecosystem, but Apple hasn't had a good record of doing that in the past. 3) would be the most pragmatic solution, but it would require Apple to make a significant investment in the build system, which has been historically slow to evolve, and there are many projects out there that accidentally build through implicit configuration that might break if such a feature is introduced.
```